contrib/mesos/pkg/executor/executor.go

```diff
@@ -212,10 +199,15 @@ func (k *KubernetesExecutor) Registered(driver bindings.ExecutorDriver,
 	}
 
 	if executorInfo != nil && executorInfo.Data != nil {
-		k.staticPodsConfig = executorInfo.Data
+		k.initializeStaticPodsSource(executorInfo.Data)
 	}
 
-	k.initialRegistration.Do(k.onInitialRegistration)
+	k.updateChan <- kubelet.PodUpdate{
```

Comment:
The other places that we write to the channel should already be guarded by
locking (or at least document that they make that assumption)

2015-09-22 17:05 GMT-04:00 Dr. Stefan Schimanski <notifications@github.com>:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/13036#discussion_r40144046>
> :
>
> >  	}
> >
> > -	k.initialRegistration.Do(k.onInitialRegistration)
> > +	k.updateChan <- kubelet.PodUpdate{
>
> We have more places where we write to the channel. Will add locking.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13036/files#r40144046>.
>